### DISCUSSION TITLE
Can etcd client which init in code can watch etcdctl add member or other members change?

### DISCUSSION THREAD

**Discussion by vintmd** [etcd-io/etcd#16238] [General]
background:
I init etcd client in code with configuration [A, B. C],
then I use the etcdctl command to add or remove node in cluster.
should I reconfigure the [A, D, E]  and restart my project re-init etcd client?
otherwise can inited etcd client can auto watch etcdctl  operation no need to restart project?

**Comment by chaochn47** [comment]
Please try experiment with `AutoSyncInterval` to a non zero value like `10s` in [clientv3.Config](https://pkg.go.dev/go.etcd.io/etcd/client/v3#Config)

etcd client will poll the cluster membership with `AutoSyncInterval` and update endpoints it connects to, then tear down the connection with the removed member and establish new connection with the newly added member.

**Reply by vintmd** [reply]
thanks! I gonna have a try, If I set AutoSyncInterval to see the source code it will period list members and update endpoints. 
so if all members which init from configuration when started changed, it also can watch and no need to restart or re-init ETCD client?
so after setting this only I need to care about is whether the old cluster configurations must change to new cluster nodes?
